scripts/itest_parallel.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Get all the variables.
+PROCESSES=$1
+TRANCHES=$2
+TEST_FLAGS=$3
+ITEST_FLAGS=$4
+
+# Create a variable to hold the final exit code.
+exit_code=0
+
+# Run commands using xargs in parallel and capture their PIDs
+pids=()
+for ((i=0; i<PROCESSES; i++)); do 
+    cmd="scripts/itest_part.sh $i $TRANCHES $TEST_FLAGS $ITEST_FLAGS"
+    echo "$i" | xargs -r -P $PROCESSES -n 1 -I $i bash -c "$cmd " & pids+=($!)
+done
+
+
+# Wait for the processes created by xargs to finish.
+for pid in "${pids[@]}"; do
+    wait $pid
+
+    # Once finished, grab its exit code.
+    current_exit_code=$?
+
+    # Overwrite the exit code if current itest doesn't return 0.
+    if [ $current_exit_code -ne 0 ]; then
+	# Only write the exit code of the first failing itest.
+	if [ $exit_code -eq 0 ]; then
+            exit_code=$current_exit_code
+	fi
+    fi
+done
+
+
+# Exit with the exit code of the first failing itest.
+if [ $exit_code -ne 0 ]; then
```

Comment:
I think we can just `exit $exit_code` in any case? If it's 0 then that should produce the same result.

Reply:
Ha yeah right. Added this check before to print some debug logs, now removed!